File: freecodecamp/daily-code-challenge/2026-04-01.js
function fixPrankNumber(arr) {
  const n = arr.length;

  // Step 1: get differences
  let diffs = [];
  for (let i = 1; i < n; i++) {
    diffs.push(arr[i] - arr[i - 1]);
  }

  // Step 2: most common difference
  let freq = {};
  for (let d of diffs) {
    freq[d] = (freq[d] || 0) + 1;
  }

  let correctDiff = Number(Object.keys(freq).reduce((a, b) =>
    freq[a] > freq[b] ? a : b
  ));

  // Step 3: try different starting points
  let best = null;

  for (let startIdx = 0; startIdx < Math.min(3, n); startIdx++) {
    let startVal = arr[startIdx] - startIdx * correctDiff;

    let candidate = [];
    for (let i = 0; i < n; i++) {
      candidate.push(startVal + i * correctDiff);
    }

    // count differences
    let changes = 0;
    for (let i = 0; i < n; i++) {
      if (candidate[i] !== arr[i]) changes++;
    }

    if (changes <= 1) return candidate;
  }

  return arr;
}

File: freecodecamp/daily-code-challenge/2026-04-01.py
from collections import Counter

def fix_prank_number(arr):
    n = len(arr)

    # Step 1: differences
    diffs = [arr[i] - arr[i-1] for i in range(1, n)]

    # Step 2: most common difference
    correct_diff = Counter(diffs).most_common(1)[0][0]

    # Step 3: try different starting points
    for start_idx in range(min(3, n)):
        start_val = arr[start_idx] - start_idx * correct_diff

        candidate = [start_val + i * correct_diff for i in range(n)]

        changes = sum(1 for i in range(n) if candidate[i] != arr[i])

        if changes <= 1:
            return candidate

    return arr